appserver/payara-appserver-modules/security-openid/src/main/java/fish/payara/security/openid/OpenIdAuthenticationMechanism.java

```diff
@@ -332,7 +333,10 @@ private AuthenticationStatus refreshTokens(HttpMessageContext httpContext, Refre
             updateContext(tokensObject);
             OpenIdCredential credential = new OpenIdCredential(tokensObject, httpContext, configuration);
             CredentialValidationResult validationResult = identityStoreHandler.validate(credential);
-            // Register session manually (if @AutoApplySession used, this would be done by its interceptor)
+            
+            // Dot not register session, as this will invalidate the currently active sessions (with all of its attributes)!
+            // httpContext.setRegisterSession(validationResult.getCallerPrincipal().getName(), validationResult.getCallerGroups());
+
             httpContext.setRegisterSession(validationResult.getCallerPrincipal().getName(), validationResult.getCallerGroups());
```

Comment:
Same line is duplicated and commented out ... was it tested? :-)

Reply:
Ah, sorry. Copied the changes from my test-env without removing the line. Will fix it.